pkg/firewall/utils/common.go

```diff
@@ -39,7 +40,45 @@ func GetIPValueType(value *string) (firewallv1beta1.IPValueType, error) {
 		return firewallv1beta1.IPValueTypeIP, nil
 	}
 
-	return firewallv1beta1.IPValueTypeVoid, fmt.Errorf("invalid match value %s", *value)
+	// Check if the value is an IP range.
+	if _, err := GetIPValueTypeRange(*value); err == nil {
+		return firewallv1beta1.IPValueTypeRange, nil
+	}
+
+	return firewallv1beta1.IPValueTypeVoid, fmt.Errorf("invalid match value IP %s", *value)
+}
+
+// GetIPValueTypeRange parses the match value and returns the type of the value.
+func GetIPValueTypeRange(s string) (firewallv1beta1.IPValueType, error) {
+	startIP, endIP, err := getIPValueRange(s)
+	if err != nil && startIP != nil && endIP != nil {
```

Comment:
```suggestion
	if err == nil && startIP != nil && endIP != nil {
```

Also, I think we can drop the checks on the startIP and endIP, because if the function return no errors (err==nil) we can safely assume that the other returned values are also valid.